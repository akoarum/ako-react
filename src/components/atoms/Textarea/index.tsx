import styled from 'styled-components'
import * as React from 'react'
import * as style from '../../../utils/style'

type Props = {
  className?: string
  name: string
  value: string
  id?: string
  minLength?: number
  maxLength?: number
  placeholder?: string
  required?: boolean
  onInput: (e: React.FormEvent<HTMLTextAreaElement>) => void
  onFocus: (e: React.FormEvent<HTMLTextAreaElement>) => void
  onBlur: (e: React.FormEvent<HTMLTextAreaElement>) => void
}

export const StyledTextarea = styled.textarea`
  box-sizing: border-box;
  border: 1px solid ${style.color.border};
  padding: 8px 12px;
  width: 100%;
  font-size: ${style.font.sizePc};
  outline: none;
  transition: border 0.3s ${style.animation.easing};

  &:focus {
    border: 1px solid ${style.color.theme};
  }

  &[aria-invalid='true'] {
    border: 1px solid ${style.color.caution};
  }

  ${style.mediaQuery.pc} {
    height: 140px;
  }

  ${style.mediaQuery.sp} {
    height: 100px;
  }
`

export const Textarea: React.FC<Props> = props => {
  return (
    <StyledTextarea
      id={props.id}
      className={props.className}
      name={props.name}
      defaultValue={props.value}
      minLength={props.minLength}
      maxLength={props.maxLength}
      placeholder={props.placeholder}
      required={props.required}
      onInput={props.onInput}
      onFocus={props.onFocus}
      onBlur={props.onBlur}
    />
  )
}

export default Textarea
