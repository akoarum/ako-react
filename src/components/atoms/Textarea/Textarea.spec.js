import React from 'react'
import renderer from 'react-test-renderer'
import Textarea from './index'

describe('Textarea', () => {
  let wrapper

  beforeEach(() => {
    wrapper = renderer.create(
      <Textarea />
    )
  })

  describe('initialize', () => {
    it('snapshot', () => {
      const tree = wrapper.toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
