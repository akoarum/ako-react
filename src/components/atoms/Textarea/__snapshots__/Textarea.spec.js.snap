// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea initialize snapshot 1`] = `
<textarea
  className="sc-bdVaJa qLBWp"
/>
`;
